attestation/src/report_data.rs

```diff
@@ -0,0 +1,161 @@
+use near_crypto::PublicKey;
+use sha3::{Digest, Sha3_384};
+
+/// Number of bytes for the report data.
+/// report_data: [u8; 64] =
+///   [version(2 bytes big endian) || sha384(TLS pub key || account pub key) || zero padding]
+const REPORT_DATA_SIZE: usize = 64;
+
+const BINARY_VERSION_OFFSET: usize = 0;
+const BINARY_VERSION_SIZE: usize = 2;
+
+const PUBLIC_KEYS_OFFSET: usize = 3;
+const PUBLIC_KEYS_SIZE: usize = 48;
+
+pub const BINARY_VERSION: BinaryVersion = BinaryVersion(1);
+
+// Compile-time assertions
+const _: () = {
+    assert!(
+        BINARY_VERSION_SIZE + PUBLIC_KEYS_SIZE <= REPORT_DATA_SIZE,
+        "Version and public key must not exceed report data size."
+    );
+    assert!(
+        BINARY_VERSION_OFFSET + BINARY_VERSION_SIZE + 1 == PUBLIC_KEYS_OFFSET,
+        "Public key offset must be after binary version."
+    );
+};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct BinaryVersion(pub u16);
+
+#[derive(Debug, Clone)]
+pub struct ReportData {
+    pub version: BinaryVersion,
+    pub tls_public_key: PublicKey,
+    pub account_public_key: PublicKey,
+}
+
+impl ReportData {
+    pub fn new(tls_public_key: PublicKey, account_public_key: PublicKey) -> Self {
+        Self {
+            version: BINARY_VERSION,
+            tls_public_key,
+            account_public_key,
+        }
+    }
+
+    /// Generates the binary representation of the report data.
+    ///
+    /// Format:
+    /// [version(2 bytes big endian) || sha384(TLS pub key || account pub key) || zero padding]
+    pub fn to_bytes(&self) -> [u8; REPORT_DATA_SIZE] {
+        let mut report_data = [0u8; REPORT_DATA_SIZE];
+
+        // Copy binary version (2 bytes, big endian)
+        let version_bytes = self.version.0.to_be_bytes();
+        report_data[BINARY_VERSION_OFFSET..BINARY_VERSION_OFFSET + BINARY_VERSION_SIZE]
+            .copy_from_slice(&version_bytes);
+
+        // Generate and copy hash of public keys (48 bytes)
+        let public_keys_hash = self.generate_public_keys_hash();
+        report_data[PUBLIC_KEYS_OFFSET..PUBLIC_KEYS_OFFSET + PUBLIC_KEYS_SIZE]
+            .copy_from_slice(&public_keys_hash);
+
+        // Remaining bytes are already zero-padded by default
+        report_data
+    }
+
+    /// Generates SHA3-384 hash of concatenated TLS and account public keys.
+    fn generate_public_keys_hash(&self) -> [u8; PUBLIC_KEYS_SIZE] {
+        let mut hasher = Sha3_384::new();
+        hasher.update(self.tls_public_key.key_data());
+        hasher.update(self.account_public_key.key_data());
+        hasher.finalize().into()
+    }
+}
+
+impl From<ReportData> for [u8; REPORT_DATA_SIZE] {
+    fn from(report_data: ReportData) -> Self {
+        report_data.to_bytes()
```

Comment:
Nit: reuse implementation below.
```suggestion
        (&report_data).into()
```

Reply:
Good point but I [decided to remove it altogether](https://github.com/near/mpc/pull/632/commits/e7bb4ef4379bfe5519b8c19a322b43c8c0372c6c) as @netrome suggested [here](https://github.com/near/mpc/pull/632#discussion_r2206682074).